test/Microsoft.ML.Tests/Scenarios/Api/AspirationalExamples.cs

```diff
@@ -60,5 +60,97 @@ public void FirstExperienceWithML()
                 PetalLength = 5.1f
             });
         }
+
+
+        [Fact]
+        public void SimpleIrisDescisionTrees()
+        {
+            var env = new ConsoleEnvironment();
+            string dataPath = "iris.txt";
+            // Create reader with specific schema.
+            var dataReader = TextLoader.CreateReader(env, dataPath,
+                ctx => (
+                    label: ctx.LoadText(0),
+                    sepalWidth: ctx.LoadFloat(1),
+                    sepalLength: ctx.LoadFloat(2),
+                    petalWidth: ctx.LoadFloat(3),
+                    petalLength: ctx.LoadFloat(4)
+                ));
+
+            var pipeline = dataReader.MakeEstimator()
+                .Append(row => (
+                    // Convert string label to key.
+                    label: row.label.Dictionarize(),
+                    // Concatenate all features into a vector.
+                    features: row.sepalWidth.ConcatWith(row.sepalLength, row.petalWidth, row.petalLength)))
+                .Append(row => (
+                    label: row.label,
+                    prediction: row.label.PredictFastTreeMultiClass(row.features)));
+
+            // Load the data into the system.
+            var data = dataReader.Read(dataPath);
+            var model = pipeline.Fit(data);
+
+            var predictions = model.Transform(dataReader.Read(testDataPath));
+            var metrics = MultiClassEvaluator.Evaluate(predictions, row => row.label, row => row.prediction);
+        }
+
+        [Fact]
+        public void TwitterSentimentAnalysis()
+        {
+            var env = new ConsoleEnvironment();
+            var dataPath = "wikipedia-detox-250-line-data.tsv";
+            // Load the data into the system.
+            var dataReader = TextLoader.CreateReader(env, dataPath,
+                    ctx => (
+                        label: ctx.LoadFloat(0),
+                        text: ctx.LoadText(1)),
+                    hasHeader: true);
+
+            var pipeline = dataReader.MakeEstimator()
+                .Append(row => (
+                    label: row.label,
```

Comment:
Is there a way to keep all columns without writing them out individually? For example, I have 1000 categorical columns and 1 text column. I would to apply TextFeaturizer to the text column but keep the 1000 categorical columns.

Reply:
For various reasons a valid schema shape is not only a value-tuple of `PipelineColumn` objects, but also recursively defined structures of value-tuples... that is, you could have a value-tuple of value-tuple of value-tuple, etc. This means that in this expression it would have been totally legal to say, `prev: row`, and then you just bring them along for free, at the cost of having to work through that `prev` identifier. This is discussed in #632, in particular, see the section "ValueTuples of PipelineColumns," and in particular the sentence "it provides a convenient way to just bring along all the inputs, if we wanted to do so, by just assigning the input tuple itself as an item in the output tuple."